therapy/query.py

```diff
@@ -528,59 +549,150 @@ def normalize(self, query: str, infer: bool = True) -> NormalizationService:
+                    response.matches[record_source].records.append(drug)
+                else:
+                    response.matches[record_source] = MatchesNormalized(
+                        records=[drug],
+                        source_meta_=self._fetch_meta(record_source.value)
+                    )
+        return response
+
+    def normalize_unmerged(
+        self, query: str, infer: bool = True
+    ) -> UnmergedNormalizationService:
+        """Return all source records under the normalized concept for the provided
+        query string.
+
+        :param str query: string to search against
+        :param bool infer: if true, try to infer namespace for IDs
+        :return: Normalized response object
+        """
+        response = UnmergedNormalizationService(
+            matches={},
+            **self._prepare_normalized_response(query)
+        )
+        if query == "":
+            return response
+        query_str = query.lower().strip()
 
-        if not matching_records:  # type: ignore
-            response["match_type"] = MatchType.NO_MATCH.value
-        return NormalizationService(**response)
+        # check merged concept ID match
+        record = self.db.get_record_by_id(query_str, case_sensitive=False, merge=True)
+        if record:
+            return self._add_normalized_records(response, record, MatchType.CONCEPT_ID)
+
+        # check concept ID match
+        record = self.db.get_record_by_id(query_str, case_sensitive=False)
+        if record:
+            return self._resolve_merge(response, query, record, MatchType.CONCEPT_ID,
+                                       self._add_normalized_records)
+
+        # check concept ID match with inferred namespace
+        if infer:
+            inferred_response = self._infer_namespace(query)
+            if inferred_response:
+                if response.warnings:
+                    response.warnings.append(inferred_response[1])
+                else:
+                    response.warnings = [inferred_response[1]]
+                return self._resolve_merge(response, query, inferred_response[0],
+                                           MatchType.CONCEPT_ID,
+                                           self._add_normalized_records)
+
+        # check other match types
+        for match_type in ITEM_TYPES.values():
+            matching_records = self._get_matches_by_type(query_str, match_type)
+
+            # attempt merge ref resolution until successful
+            for match in matching_records:
+                assert match is not None
+                record = self.db.get_record_by_id(match["concept_id"], False)
+                if record:
+                    match_type_value = MatchType[match_type.upper()]
+                    return self._resolve_merge(response, query, record,
+                                               match_type_value,
+                                               self._add_normalized_records)
+        return response
```

Comment:
Just from a quick glance, this seems to also be in `normalize`. Should separate out into a new method and call it in  `normalize` + here